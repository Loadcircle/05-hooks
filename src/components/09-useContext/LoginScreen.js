import React, {useContext} from 'react';
import { UserContext } from './UserContext';

export const LoginScreen = () => {

    const {setUser} = useContext(UserContext);

    const newUser = {
        id: 4,
        name: 'Jesus Milano',
        email: 'jesus.milano@valiapro.com'
    }
    return (
        <div>
            <h1>Login Screen</h1>
            <hr/>

            <button 
                className="btn btn-primary"
                onClick={()=>{ setUser(newUser) }}
            >
                Login
            </button>
        </div>
    )
}
