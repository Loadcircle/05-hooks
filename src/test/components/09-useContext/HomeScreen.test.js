import React from 'react';
import { UserContext } from '../../../components/09-useContext/UserContext';
const { shallow, mount } = require("enzyme");
const { HomeScreen } = require("../../../components/09-useContext/HomeScreen");

describe('Pruebas en HomeScreen', ()=>{
    
    const user = {
        name: 'Jesus',
        email: 'Jesusmilano96@gmail.com'
    }

    const wrapper = mount(
        <UserContext.Provider value={{
            user
        }}>
            <HomeScreen />
        </UserContext.Provider>
    )

    test('Snapshot', ()=>{

        expect(wrapper).toMatchSnapshot();

    });
    

})