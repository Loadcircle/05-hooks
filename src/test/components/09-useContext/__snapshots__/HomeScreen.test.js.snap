// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en HomeScreen Snapshot 1`] = `
<div>
  <h1>
    Home Screen
  </h1>
  <hr />
  <pre>
    {
   "name": "Jesus",
   "email": "Jesusmilano96@gmail.com"
}
  </pre>
</div>
`;
